webapp/src/components/testing/TestSteps.vue

```diff
@@ -58,6 +58,34 @@
       </div>
     </div>
 
+    <div v-if="confusionMatrix && confusionMatrix.matrix.length > 0" class="p-4 mx-auto lg:w-1/2 h-full bg-white dark:bg-slate-950 rounded-md">
+      <h4 class="p-4 text-lg font-semibold text-slate-500 dark:text-slate-300">
+        Confusion Matrix
+      </h4>
+      <table class="min-w-full divide-y divide-slate-600 dark:divide-slate-400 text-center">
+        <thead>
+          <tr>
+            <th class="pl-6 py-3 text-xs font-medium text-gray-800 dark:text-gray-200 uppercase tracking-wider text-center border-r-gray-600 dark:border-r-gray-400 border-r-2 diagonal-header">                  
+              <span class="">Label \ Prediction</span>
+            </th>              
+            <th v-for="(label, index) in confusionMatrix.matrix[0]" :key="'header-' + index" class="px-6 py-3 text-xs font-medium text-gray-800 dark:text-gray-200 uppercase tracking-wider">
```

Comment:
I feel as the cells are too big, too much space around the various data. consider removing much of the `px-6 py-3` classes on the various `th` & `td` to make it more tightly packed which would be more readable and scales much better on smaller screen.

Reply:
I did what I could in order to make it more compact. I don't think it is perfect, personnaly I prefer it with space, but that is for sure more readable